lib/stages/HFBAccumulate.hpp

```diff
@@ -60,16 +64,55 @@ class HFBAccumulate : public kotekan::Stage {
     /// Primary loop to wait for buffers, dig through data,
     /// stuff packets lather, rinse and repeat.
     void main_thread() override;
+
+private:
     /// Copy the first frame of the integration
-    void init_first_frame(float* input_data, float* sum_data, const uint32_t in_buffer_ID);
+    void init_first_frame(float* input_data, float* sum_data, const uint32_t in_frame_id);
     /// Add a frame to the integration
-    void integrate_frame(float* input_data, float* sum_data, const uint32_t in_buffer_ID);
+    void integrate_frame(float* input_data, float* sum_data, const uint32_t in_frame_id);
     /// Normalise frame after integration has been completed
-    void normalise_frame(float* sum_data, const uint32_t in_buffer_ID);
+    void normalise_frame(float* sum_data, const uint32_t in_frame_id);
+
+    // NOTE: Annoyingly this can't be forward declared, and defined fully externally
+    // as the std::deque needs the complete type
+    /**
+     * @class internalState
+     * @brief Hold the internal state of a gated accumulation.
+     **/
+    struct internalState {
```

Comment:
Given that we probably won't ever have gated HFB acq's you don't really need the internalState. You could just put all of it's state on the main class. Not one to do now, but probably worth noting for future cleanups.